File: src/main/java/guru/xgm/angelscript/openapi/fileTypes/AngelScriptFileType.java
package guru.xgm.angelscript.openapi.fileTypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import guru.xgm.angelscript.icons.AngelScriptIcons;
import guru.xgm.angelscript.lang.AngelScriptLanguage;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class AngelScriptFileType extends LanguageFileType {

    public static final AngelScriptFileType INSTANCE = new AngelScriptFileType();

    private AngelScriptFileType() {
        super(AngelScriptLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "AngelScript";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "AngelScript language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "as";
    }

    @Override
    public @NotNull Icon getIcon() {
        return AngelScriptIcons.FILE;
    }
}

File: src/main/kotlin/raft/war/language/angelscript/AngelScriptFileType.kt
package raft.war.language.angelscript

import com.intellij.openapi.fileTypes.LanguageFileType
import raft.war.language.angelscript.icons.AngelScriptIcons
import javax.swing.Icon

class AngelScriptFileType private constructor() : LanguageFileType(AngelScriptLanguage.instance) {
    override fun getName(): String = AngelScriptLanguage.NAME

    override fun getDescription(): String = AngelScriptLanguage.NAME

    override fun getDefaultExtension(): String = EXTENSION

    override fun getIcon(): Icon = AngelScriptIcons.FILE

    @Suppress("CompanionObjectInExtension")
    companion object {
        val instance: AngelScriptFileType = AngelScriptFileType()
        const val EXTENSION: String = "as"
    }
}
